packages/integration_test/example/android/app/build.gradle

```diff
@@ -29,6 +29,10 @@ apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
+    // Conditional for compatibility with AGP <4.2.
+    if (project.android.hasProperty("namespace")) {
+        namespace 'com.example.integration_test_example'
+    }
```

Comment:
Apparently I didn't save the comment that was supposed to go with that overall comment:

Could we make *just* this change in one PR, to make it as safe as possible to cherry-pick, and do the rest in a second PR that we don't cherry-pick? I'm slightly worried that the AGP update here might cause unexpected issues for clients (since IIRC we've hit issues in the past where we thought that wouldn't affect clients and then it did, but maybe I'm misremembering).

Reply:
https://github.com/flutter/flutter/pull/125657